modules/tinymce/src/core/main/ts/fmt/ApplyFormat.ts

```diff
@@ -191,6 +192,11 @@ const applyFormat = (ed: Editor, name: string, vars?: FormatVars, node?: Node |
         if (FormatUtils.isSelectorFormat(format)) {
           const found = applyNodeStyle(formatList, node);
 
+          // TINY-6567 Include the last node in the selection
+          if (!NodeType.isElement(node) && hasBlockChildren(node.parentNode)) {
```

Comment:
Maybe change this if to
`NodeType.isText(node) && hasBlockChildren(node.parentNode) && nodes[node.length - 1] === node`
or
`NodeType.isText(node) && hasBlockChildren(node.parentNode) && Arr.last(nodes).is(node)`

Since it's supposed to check the last node only and if it's not an element it can only be text node, comment or document but we don't want to process the last two. But the RangeWalk.walk will fire the callback multiple times so it's not really the truth either. But it will limit the scope of this apply.

I'm also a bit unsure what impact this would have wouldn't it double apply the node style if a match is made in the middle of the selection. I guess a selector format is always just updating attributes and such an operation is idempotent.

Reply:
Instead of always check the last element it's better to used Arr.contains because if the user types something like 
(<br/>test) or (test<br/>) the li parent won't be selected, also I update some tests for those cases.